src/Navigation.tsx

```diff
@@ -79,13 +80,16 @@ function MainStack({navigation}: Props) {
   const headerHeightIncludingSafeArea = useHeaderHeight()
 
   const {loginState} = useContext(AuthContext)
+  const {user} = useContext(UserContext)
 
   useEffect(() => {
     if (loginState !== LoginState.LoggedOut) {
       navigation.navigate(SCREENS.HOME)
     }
   }, [loginState])
 
+  useEffect(() => {}, [user])
```

Comment:
?

Reply:
Means that the functional component will rerender if the user is set. Which I want in the case of loading up offline asynchronous data. In this case it will ensure that header displays the users name even if the app is currently in a logging in state.

Reply:
Oh right. Does this not work without this use effect? The provider should be making it so the children re-render on a prop change.

Reply:
> The provider should be making it so the children re-render on a prop change.

This effect is required. It's used by the navigation header to show the setting icon in the header if the user gets loaded from the cache.